superset/advanced_data_type/plugins/eml_path.py

```diff
@@ -0,0 +1,81 @@
+import re
+from typing import Any, List
+
+from sqlalchemy import Column
+from superset.advanced_data_type.types import (
+    AdvancedDataType,
+    AdvancedDataTypeRequest,
+    AdvancedDataTypeResponse,
+)
+from superset.utils.core import FilterOperator, FilterStringOperators
+
+# Operator Sets
+
+EQUAL_NULLABLE_OPERATOR_SET = [
+            FilterStringOperators.EQUALS,
+            FilterStringOperators.NOT_EQUALS,
+            FilterStringOperators.IN,
+            FilterStringOperators.NOT_IN,
+            FilterStringOperators.IS_NOT_NULL,
+            FilterStringOperators.IS_NULL,
+        ]
+
+def translate_filter_func(col: Column, operator: FilterOperator, values: List[Any]) -> Any:
+    """
+    Convert a passed in column, FilterOperator and
+    list of values into an sqlalchemy expression
+    """
+    return_expression: Any
+    if operator == FilterOperator.IN.value:
+        return_expression = col.in_(values)
+    elif operator == FilterOperator.NOT_IN.value:
+        return_expression = ~(col.in_(values))
+    elif operator == FilterOperator.IS_NULL.value:
+        return_expression = col == None
+    elif operator == FilterOperator.IS_NOT_NULL.value:
+        return_expression = col != None
+    elif len(values) == 1:
+        value = values[0]
+        if operator == FilterOperator.EQUALS.value:
+            return_expression = col == value
+        if operator == FilterOperator.NOT_EQUALS.value:
+            return_expression = col != value
+        if operator == FilterOperator.LIKE.value:
+            return_expression = col.like(value)
+        if operator == FilterOperator.ILIKE.value:
+            return_expression = col.ilike(value)
+    return return_expression
+
+
+def eml_path_func(req: AdvancedDataTypeRequest) -> AdvancedDataTypeResponse:
+    """
+    Convert a passed in AdvancedDataTypeRequest to an AdvancedDataTypeResponse
+    """
+    resp: AdvancedDataTypeResponse = {
+        "values": [],
+        "error_message": "",
+        "display_value": "",
+        "valid_filter_operators": EQUAL_NULLABLE_OPERATOR_SET,
+    }
+    if req["values"] == [""]:
+        resp["error_message"] = "EML_path must not be empty"
+        return resp
+    for val in req["values"]:
+        string_value = str(val)
+        if re.search("^((C|N)BS_EMAIL:\/\/)([0-9]{4}\/[0-9]{2}\/[0-9]{2}\/eml\/(c|n)bs\/).*(\.eml\.cart)$", string_value):
```

Comment:
This should be a constant so it can be reused in the error message.

Reply:
Edited in new commit.